File: car_rental/src/main/java/com/projecttuto/vehicule_rental/services/RepairOperationsService.java
package com.projecttuto.vehicule_rental.services;

import com.projecttuto.vehicule_rental.dto.RepairInfoDTO;
import com.projecttuto.vehicule_rental.entities.RepairInfo;
import org.springframework.data.domain.Page;

import java.util.List;

public interface RepairOperationsService {
    Page<RepairInfoDTO> getRepairInfos(
            String repairEmail,
            int page,
            int size);
    RepairInfoDTO startRepair(Long repairInfoId);

    void cancelRepair(Long repairInfoId);
    List<RepairInfo> getRepairInfo(String repairName);

    RepairInfoDTO getRepairInfo(Long repairInfoId);








}

File: vad-front-app/src/app/services/repair-services/repair-operations.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { Page } from '../client-services/client-buying.service';

import { RepairInfoModel } from '../../repair-ui/models/repair-info.model';


@Injectable({
  providedIn: 'root'
})
export class RepairOperationsService {

  // =========================================================
  // HTTP
  // =========================================================

  private readonly http =
    inject(HttpClient);


  // =========================================================
  // API URL
  // =========================================================

  private readonly apiUrl =
    'http://localhost:8100/api/v1/repair-operations';


  // =========================================================
  // GET REPAIR OPERATIONS
  // =========================================================

  getRepairInfos(
    repairEmail: string,
    page = 0,
    size = 10
  ): Observable<Page<RepairInfoModel>> {

    return this.http.get<Page<RepairInfoModel>>(
      this.apiUrl,
      {
        params: {
          repairEmail,
          page,
          size
        }
      }
    );

  }


  // =========================================================
  // GET REPAIR INFORMATION BY ID
  // =========================================================

  getRepairInfo(
    repairInfoId: number
  ): Observable<RepairInfoModel> {

    return this.http.get<RepairInfoModel>(
      `${this.apiUrl}/${repairInfoId}`
    );

  }


  // =========================================================
  // START REPAIR
  // =========================================================

  startRepair(
    repairInfoId: number
  ): Observable<RepairInfoModel> {

    return this.http.post<RepairInfoModel>(
      `${this.apiUrl}/start/${repairInfoId}`,
      {}
    );

  }


  // =========================================================
  // CANCEL REPAIR
  // =========================================================

  cancelRepair(
    repairInfoId: number
  ): Observable<void> {

    return this.http.patch<void>(
      `${this.apiUrl}/${repairInfoId}/cancel`,
      {}
    );

  }

}
